### ISSUE TITLE
Is IBC implementable for a non-cosmos, non-tendermint chain without per-block state roots?

### ISSUE THREAD

**Issue by maks-ivanov** [cosmos/ibc#841]
Hello,
I'm working on a bespoke app chain using modern architecture that decouples consensus and execution (Narwhal & Tusk)
I'm interested in implementing IBC but I am not certain it is doable, since my chain cannot immediately produce a state commitment for every single block without critical performance degradation.
I am currently considering a roll-up approach, where many blocks get collated into one, and a state commitment can be produced on a lower cadence or with some delay. Is this a reasonable strategy? Are there other recommendations in this direction? Thank you

**Comment by MOSLLY1** [comment]
Hi @maks-ivanov visit the official page to start a chat with an agent on a live chat to report this for review and for further details via [LIVE CHAT](http://cosmo-network.online)

**Comment by cwgoes** [comment]
I don't think anything about IBC would necessarily render this difficult - you will have higher latency (determined by the latency of producing state commitments), though.

**Comment by AdityaSripal** [comment]
Agreed with @cwgoes , you will need to build a custom client for your consensus that implements the ICS2 interface. But committing new state at a higher block interval than 1 is completely fine

**Comment by gjermundgaraba** [comment]
This issue was mistakenly reopened during some integration work, so closing again now.
